radis/api/kuruczapi.py

```diff
@@ -0,0 +1,451 @@
+            jupper: upper J
+            ielem:  atomic number (e.g., Fe=26)
+            iion:  ionized level (e.g., neutral=1, singly)
+            gamRad: log of gamma of radiation damping (s-1) #(https://www.astro.uu.se/valdwiki/Vald3Format)
+            gamSta: log of gamma of Stark damping (s-1)
+            gamvdW:  log of (van der Waals damping constant / neutral hydrogen number) (s-1)
+        """
+        with open(kuruczf) as f:
+            lines = f.readlines()
+        (
+            wlnmair,
+            loggf,
+            species,
+            elower,
+            jlower,
+            labellower,
+            eupper,
+            jupper,
+            labelupper,
+            gamRad,
+            gamSta,
+            gamvdW,
+            ref,
+            NLTElower,
+            NLTEupper,
+            isonum,
+            hyperfrac,
+            isonumdi,
+            isofrac,
+            hypershiftlower,
+            hypershiftupper,
+            hyperFlower,
+            hypernotelower,
+            hyperFupper,
+            hypternoteupper,
+            strenclass,
+            auto,
+            landeglower,
+            landegupper,
+            isoshiftmA,
+        ) = (
+            np.zeros(len(lines)),
+            np.zeros(len(lines)),
+            np.array([""] * len(lines), dtype=object),
+            np.zeros(len(lines)),
+            np.zeros(len(lines)),
+            np.zeros(len(lines)),
+            np.zeros(len(lines)),
+            np.zeros(len(lines)),
+            np.zeros(len(lines)),
+            np.zeros(len(lines)),
+            np.zeros(len(lines)),
+            np.zeros(len(lines)),
+            np.zeros(len(lines)),
+            np.zeros(len(lines)),
+            np.zeros(len(lines)),
+            np.zeros(len(lines)),
+            np.zeros(len(lines)),
+            np.zeros(len(lines)),
+            np.zeros(len(lines)),
+            np.zeros(len(lines)),
+            np.zeros(len(lines)),
+            np.zeros(len(lines)),
+            np.zeros(len(lines)),
+            np.zeros(len(lines)),
+            np.zeros(len(lines)),
+            np.zeros(len(lines)),
+            np.zeros(len(lines)),
+            np.zeros(len(lines)),
+            np.zeros(len(lines)),
+            np.zeros(len(lines)),
+        )
+        ielem, iion = np.zeros(len(lines), dtype=int), np.zeros(len(lines), dtype=int)
+
+        for i, line in enumerate(lines):
+            wlnmair[i] = float(line[0:11])
+            loggf[i] = float(line[11:18])
+            species[i] = str(line[18:24])
+            ielem[i] = int(species[i].split(".")[0])
+            iion[i] = int(species[i].split(".")[1]) + 1
+            elower[i] = float(line[24:36])
+            jlower[i] = float(line[36:41])
+            eupper[i] = float(line[52:64])
+            jupper[i] = float(line[64:69])
+            gamRad[i] = float(line[80:86])
+            gamSta[i] = float(line[86:92])
+            gamvdW[i] = float(line[92:98])
```

Comment:
There are usually much faster ways to parse a file than enumerate through all lines. 

See how we do in the HITRAN parser